src/containers/bookslist.js

```diff
@@ -18,12 +18,12 @@ const BooksList = props => {
     const result = [];
     if (category === FILTERS.all) {
       books.forEach((item, i) => {
-        result.push(<Book key={item.id} book={item} delBook={() => deleteBook(i)} />);
+        result.push(<Book key={item.id} book={item} delBook={() => handleRemoveBook(i)} />);
       });
     } else {
       books.forEach((item, i) => {
         if (item.category === category) {
-          result.push(<Book key={item.id} book={item} delBook={() => deleteBook(i)} />);
+          result.push(<Book key={item.id} book={item} delBook={() => handleRemoveBook(i)} />);
         }
       });
```

Comment:
[Required] Instead of this, we can simply filter books based on the selected category and rendered the filtered books with map function.

Reply:
Hello Suman, thanks for your kind review! regarding this point, I did this implementation because I need the index in the array stored in the redux store, with filter a different index would be provided.

Reply:
The use of methods like `.filter`, `.map`, `.slice` are React friendly since they return copies of other data based on a  specify rule, React convention is to not mutate but return a copy of the old data. I am not sure that using his approach would worsen your time complexity, I still think both are approach are O(n)